core/src/main/scala/org/http4s/rho/AuthedRhoService.scala

```diff
@@ -0,0 +1,65 @@
+package org.http4s
+package rho
+
+import org.http4s.rho.bits.PathAST.TypedPath
+import org.http4s.AuthedService
+import org.log4s.getLogger
+import shapeless.{HNil, HList}
+
+/** Constructor class for defining routes
+  *
+  * The [[AuthedRhoService]] provides a convenient way to define routes in a style
+  * similar to scalatra etc by providing implicit conversions and an implicit
+  * [[CompileService]] inside the constructor.
+  *
+  * {{{
+  *   val srvc = new AuthedRhoService[U] {
+  *     POST / "foo" / pathVar[Int] +? param[String]("param") |>> { (req: Request, p1: Int, param: String) =>
+  *       val authInfo: U = getAuth(req)
+  *       ...
+  *     }
+  *   }
+  *
+  * }}}
+  *
+  * @tparam U authInfo type for this service.
+  * @param routes Routes to prepend before elements in the constructor.
+  *
+  */
+class AuthedRhoService[U](routes: Seq[RhoRoute[_ <: HList]] = Vector.empty)
+  extends bits.MethodAliases
+    with bits.ResponseGeneratorInstances
+    with RoutePrependable[AuthedRhoService[U]]
+    with EntityEncoderInstances
+{
+  final private val serviceBuilder = AuthedServiceBuilder[U](routes, ev)
+
+  final protected val logger = getLogger
+
+  final implicit protected def compileService: CompileService[RhoRoute.Tpe] = serviceBuilder
+
+  /** Create a new [[AuthedRhoService]] by appending the routes of the passed [[AuthedRhoService]]
+    *
+    * @param other [[AuthedRhoService]] whos routes are to be appended.
+    * @return A new [[AuthedRhoService]] that contains the routes of the other service appended
+    *         the the routes contained in this service.
+    */
+  final def and(other: AuthedRhoService[U]): AuthedRhoService[U] = new AuthedRhoService(this.getRoutes ++ other.getRoutes)
```

Comment:
This is going to have problems with the referential equality of `AttributeKey`: when you define a route, you use the `getAuth` of the DSL instance where it is defined. However, the conversion of the routes to an `AuthedService` uses the key from the DSL instance performing the compilation, resulting in different keys.